RIP-3 Nft Bank/RIP-3 Nft Bank.md

```diff
@@ -0,0 +1,85 @@
+    -   bytes32 - password for locking the account
+    -   address[] backupAddresses- backUp accounts in case the main account's private keys are lost
+#### Bank.sol-
+Imports:
+-   IERC721Upgradeable and IERC1155Upgradeable which allows the contract to make transfer calls.
+-   ERC165CheckerUpgradeable which gives the contract the ability to check supported interfaces.
+
+Inherits:
+-   ERC1155HolderUpgradeable and ERC721HolderUpgradeable which gives the contract the ability to receive and store NFT's.
+-   AccessControlUpgradeable which allows the contract to grant account access to backup addresses.
+-   PausableUpgradeable which provides the functionality of locking functions in case a bug is discovered.
+-   IBank and IBankStorage interfaces.
+
+Global variables:
+-   (address => LibBank.SecurityInfo) accountInfo
+-   uint64 maxSecurityTimeout- a blockNumber which represents the length of time required before a backup address will have access to an account
+-   uint64 lastPaused - blockNumber of when the contracts have been frozen
+
+Functions:
+
+Note: All mutative functions will have a modifier that requires that either msg.sender == the affected account's address or that the current blockNumber - lastAccessed variable is over a certain time frame and that msg.sender == one of the backup addresses. And all mutative functions will reset the lastAccessed blockNumber of an address to the current blockNumber, but not if the functions were accessed by the backup addresses.
+
+-   depositBatch function to deposit multiple assets. This function takes in an array of assetData structs, an array of uint256 deposit amounts and an address.
+    -   checks if the user possesses the amount of tokens they want to deposit.
+    -   calls the deposit function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to send the item to Bank.sol.
+-   withdrawBatch function to withdraw multiple assets. This function takes in an array of assetData structs, an array of uint256 withdraw amounts and an address.
+    -   requires that stored amount of the item is greater than or equal to the amount to withdraw.
+    -   calls the withdraw function in BankStorage.sol to update the storedAmounts mapping.
+    -   calls a transfer function to retrieve the item from the bank.
+-   addBackup function to add a back up address in case one loses their keys.
+-   removeBackup function to remove an address from the backupAddresses array.
+-   setPassword function which sets an account password required to lock and unlock an account.
+-   lock function which requires a password and signature to set the accountLocked boolean to true or false.
+-   tokenAmount function to call the tokenAmount function in BankStorage.sol
+-   hashAssetId internal function which creates a hash of the item's contract address and token Id.
+
+#### BankStorage.sol
+Mappings:
+-   (bytes4 assetIdHash => uint256 tokenAmount) storedAmounts 
```

Comment:
I think that its better to do:

mapping(address => mapping(tokenId => amount)) assetData

so you have assetData[contractAddress][tokenId] = amount;

My reasoning is that you don't have to call keccak256("contractAddress","tokenId") to store or update the amount data. it's a simple look up storedAmounts[address][tokenId] instead.